### ISSUE TITLE
AWS::Partition pseudo parmeter causes critical lint error

### ISSUE THREAD

**Issue by nskoufis** [martysweet/cfn-lint#178]
Hi again!

Another Pseudo parameter that doesn't appear to be supported, discovered while building some API Gateway integrations.

Minimum working example:

```json
{
  "Resources": {
    "GatewayMethodGet": {
      "Type": "AWS::ApiGateway::Method",
      "Properties": {
        "HttpMethod": "GET",
        "RequestParameters": {
          "method.request.path.sample": true
        },
        "ResourceId": "abc123",
        "RestApiId": "abc123",
        "ApiKeyRequired": true,
        "AuthorizationType": "NONE",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region"
                },
                ":lambda:path/2015-03-31/functions/",
                "someFunctionName",
                "/invocations"
              ]
            ]
          }
        },
        "MethodResponses": []
      }
    }
  }
}
```

Expected behaviour: validates with no errors
Actual behaviour: 1 critical validation error:
```
Resource: Resources > GatewayMethodGet > Properties > Integration > Uri
Message: Referenced value AWS::Partition does not exist
Documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html
```
Note that the referenced documentation page doesn't list `AWS::Partition` as a pseudo parameter. However the pseudo parameter documentation that is linked to from there does list it. I'll be submitting a PR to rectify this.

PR for this one incoming...
